gufe/settings/models.py

```diff
@@ -108,23 +90,15 @@ class ThermoSettings(SettingsBaseModel):
        No checking is done to ensure a valid thermodynamic ensemble is
        possible.
     """
-
-    temperature: FloatQuantity["kelvin"] = Field(None, description="Simulation temperature, default units kelvin")
-    pressure: FloatQuantity["standard_atmosphere"] = Field(
-        None, description="Simulation pressure, default units standard atmosphere (atm)"
-    )
+    # TODO: do we actually want None to be valid here?
+    temperature: _TemperatureQuantity | None = Field(None, description="Simulation temperature, default units kelvin")  # TODO: make type equiv of FloatQuantity["kelvin"] =
```

Comment:
@IAlibay @hannahbaumann 
In general, which of the following do you prefer? (or is it on a very case-by-case basis?)

In increasing strictness:

1. Any temperature unit is fine (as determined by [openff.toolkit.Quantity.is_compatible_with()](https://docs.openforcefield.org/projects/units/en/stable/api/generated/openff.units.Quantity.html?highlight=is_compatible_with#openff.units.Quantity.is_compatible_with())
2. Require the precise unit and convert if possible (e.g. convert angstroms to units)
3. Require the precise unit and throw a validation error otherwise.

Reply:
Case 1 is fine (in protocols we always do `to(unit)` as best practices, case 2 could be nice and would make things a bit more resilient, case 3 would be terrible.